### PR TITLE
Support panel fastapi behind reverse proxy

### LINKED ISSUE #7277: panel fastapi: ValueError: too many values to unpack (expected 2)
I tried using the simple panel fastapi hello world example with `panel==1.5.0rc2` and `bokeh_fastapi==0.1.0`.

I get

```bash
$ uvicorn script:app
INFO:     Started server process [36967]
INFO:     Waiting for application startup.
INFO:     Application startup complete.
INFO:     Uvicorn running on http://127.0.0.1:8000 (Press CTRL+C to quit)
INFO:     ::ffff:172.20.0.233:0 - "GET / HTTP/1.1" 200 OK
INFO:     ::ffff:172.20.0.233:0 - "GET /mini-coi.js HTTP/1.1" 404 Not Found
INFO:     ::ffff:172.20.0.233:0 - "GET /panel HTTP/1.1" 500 Internal Server Error
ERROR:    Exception in ASGI application
Traceback (most recent call last):
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/uvicorn/protocols/http/h11_impl.py", line 406, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/uvicorn/middleware/proxy_headers.py", line 70, in __call__
    return await self.app(scope, receive, send)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/fastapi/applications.py", line 1054, in __call__
    await super().__call__(scope, receive, send)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/applications.py", line 123, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 186, in __call__
    raise exc
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/middleware/errors.py", line 164, in __call__
    await self.app(scope, receive, _send)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/middleware/exceptions.py", line 62, in __call__
    await wrap_app_handling_exceptions(self.app, conn)(scope, receive, send)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/routing.py", line 758, in __call__
    await self.middleware_stack(scope, receive, send)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/routing.py", line 778, in app
    await route.handle(scope, receive, send)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/routing.py", line 299, in handle
    await self.app(scope, receive, send)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/routing.py", line 79, in app
    await wrap_app_handling_exceptions(app, request)(scope, receive, send)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 64, in wrapped_app
    raise exc
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/_exception_handler.py", line 53, in wrapped_app
    await app(scope, receive, sender)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/starlette/routing.py", line 74, in app
    response = await func(request)
               ^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 278, in app
    raw_response = await run_endpoint_function(
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/fastapi/routing.py", line 191, in run_endpoint_function
    return await dependant.call(**values)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/bokeh_fastapi/handler.py", line 122, in get
    session = await self.get_session(request, bokeh_session_id)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/bokeh_fastapi/handler.py", line 109, in get_session
    session = await self.application_context.create_session_if_needed(
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/bokeh/server/contexts.py", line 240, in create_session_if_needed
    self._application.initialize_document(doc)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/panel/io/application.py", line 126, in initialize_document
    super().initialize_document(doc)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/bokeh/application/application.py", line 190, in initialize_document
    h.modify_document(doc)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/bokeh/application/handlers/function.py", line 140, in modify_document
    self._func(doc)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/panel/io/application.py", line 78, in _eval_panel
    doc = as_panel(panel)._modify_doc(server_id, title, doc, location)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/panel/viewable.py", line 318, in _modify_doc
    return self.server_doc(doc, title, location) # type: ignore
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/panel/viewable.py", line 1017, in server_doc
    self._add_location(doc, location, model)
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/panel/viewable.py", line 332, in _add_location
    loc = state.location
          ^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/panel/io/state.py", line 1040, in location
    loc = Location.from_request(self.curdoc.session_context.request)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/jovyan/repos/private/awesome-panel/.venv/lib/python3.11/site-packages/panel/io/location.py", line 80, in from_request
    params['hostname'], params['port'] = request.host.split(':')
    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ValueError: too many values to unpack (expected 2)
```

## Reproduce

```python
import panel as pn
from fastapi import FastAPI
from panel.io.fastapi import add_application

app = FastAPI()


@app.get("/")
async def read_root():
    return {"Hello": "World"}


@add_application("/panel", app=app, title="My Panel App")
def create_panel_app():
    slider = pn.widgets.IntSlider(name="Slider", start=0, end=10, value=3)
    return slider.rx() * "⭐"
```

```bash
uvicorn script:app
```

## Problem

The challenge is that I'm running this in a VS Code (Code Server)  terminal on a JupyterHub and some proxying is taking place. Thus the host is not as you would expect.

image

### PR DESCRIPTION
**Pull request by MarcSkovMadsen**
Partial fix for #7277

An attempt to fix or make it easier to fix #7277.

### CODE REVIEW

**Comment by MarcSkovMadsen** [comment]
The issue currently blocking is then that it tries to find the static assets at `/static/...`

image

This would also be a problem for anyone trying to deploy this in a docker container and not at `/`.

I will need help to solve this.

File: panel/io/location.py [hunk 1]
```diff
@@ -25,6 +25,46 @@
     from bokeh.server.contexts import BokehSessionContext
     from pyviz_comms import Comm
 
+def _get_location_params(protocol: str|None, host: str| None, uri: str| None)->dict:
```
Comment (philippjfr) [inline panel/io/location.py:28]: Completely unclear to me what you have changed here, and how this relates to reverse proxies.

**Comment by philippjfr** [comment]
Doubtful this can/should be fixed in Panel. The improved handling probably has to be in bokeh-fastapi.

File: panel/io/location.py (hunk 1 above)
Comment (philippjfr) [reply panel/io/location.py:28]: Sorry, see the original issue now.

**Comment by philippjfr** [comment]
The location fix looks fine to me.

**Comment by philippjfr** [comment]
pre-commit.ci autofix

**Comment by MarcSkovMadsen** [comment]
> Doubtful this can/should be fixed in Panel. The improved handling probably has to be in bokeh-fastapi.

Do I need to do anything here?

**Review by philippjfr — APPROVED** [review]

**Pull request merged**